lib_xua/src/core/endpoint0/xua_endpoint0.c

```diff
@@ -866,15 +868,24 @@ void XUA_Endpoint0_loop(XUD_Result_t result, USB_SetupPacket_t sp, chanend c_ep0
 
                         /* If running in application mode stop audio */
                         /* Don't interupt audio for save and restore cmds */
-                        if ((DFU_IF == INTERFACE_NUMBER_DFU) && (sp.bRequest != XMOS_DFU_SAVESTATE) &&
-                            (sp.bRequest != XMOS_DFU_RESTORESTATE))
+                        if (!DFU_mode_active
+                                && (sp.bRequest != XMOS_DFU_SAVESTATE)
+                                && (sp.bRequest != XMOS_DFU_RESTORESTATE))
                         {
-                            assert((c_audioControl != null) && msg("DFU not supported when c_audioControl is null"));
+                            /* Send STOP_AUDIO_FOR_DFU command. This will either pass through
+                             * buffering system (i.e. ep_buffer/decouple) if the device has USB audio
+                             * channels. Otherwise this directly interacts with AudioHub
+                             * This command needs to be sent such that AudioHub runs the DFUHandler()
+                             * task - in the case where AudioHub is running on tile[0] i.e the
+                             * flash tile and the USB code (i.e this task) are running on separate
+                             * tiles. It also means that Flash pins can be shared with "audio" pins.
+                             */
+                            assert((c_aud_ctl != null) && msg("DFU not supported when c_aud_ctl is null"));
```

Comment:
I think we don't define XASSERT_ENABLE_ASSERTIONS=0? If so this could end badly at runtime if someone made a mistake in config. Mind you it will anyway I guess.

Reply:
We set it in lib_build_info.cmake (in lib_xua)